test/tests/attachmentsTest.js

```diff
@@ -279,6 +279,29 @@ describe("Zotero.Attachments", function() {
 		});
 	});
 	
+	
+	describe("#importFromURL()", function () {
+		it("should download a PDF from a JS redirect page", async function () {
+			this.timeout(65e3);
+			
+			let stub = sinon.stub(Zotero.Attachments, 'downloadFile');
+			stub.callsFake(Zotero.Attachments.downloadPDFViaBrowser.bind(Zotero.Attachments));
+
+			var item = await Zotero.Attachments.importFromURL({
+				libraryID: Zotero.Libraries.userLibraryID,
+				url: 'https://zotero-static.s3.amazonaws.com/test-pdf-redirect.html',
+				contentType: 'application/pdf'
+			});
+			
+			assert.isTrue(item.isPDFAttachment());
```

Comment:
We should test that the PDF was actually saved to disk (by stealing a couple lines from `_enforcePDF()`).